src/test/shared/schema/schemaTypes.ts

```diff
@@ -0,0 +1,25 @@
+/*!
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export type JSONValue = string | boolean | number | JSONValue[] | JSONObject
+
+export interface JSONObject {
+    [key: string]: JSONValue
+}
+
+/**
+ * Traverse through the initial JSON object, visiting all of the properties.
+ * Only suitable for accessing JSON objects.
+ * @param initialObject the object you want to start the traversal at
+ * @param properties the properties you want to visit and traverse into
+ * @returns The location in initialObject after visiting all of properties
+ */
+export function unmarshal(initialObject: JSONObject, ...properties: string[]) {
```

Comment:
This is just indexing at a path yeah?

Here's a pretty strongly-typed implementation (just for fun):
```ts
type ObjectPath<T extends Record<string, any>, K extends PropertyKey[] = PropertyKey[]> = K extends []
    ? [keyof T]
    : IndexObject<T, K> extends Record<string, any>
    ? K | [...K, keyof IndexObject<T, K>]
    : K

type IndexObject<T extends Record<string, any>, K> = K extends [infer L, ...infer R]
    ? L extends keyof T
        ? Index<T[L], R>
        : never
    : T

type Index<T, K> = T extends Record<string, any> ? IndexObject<T, K> : T extends undefined ? undefined : never

function get<T extends Record<string, any>, P extends ObjectPath<T>>(
    obj: T,
    path: P
): IndexObject<T, P> {
    const next = obj[path[0] as keyof T]

    if (path.length === 1 || next === undefined) {
        return next
    }

    if (typeof next !== 'object') {
        throw new Error(`Cannot index primitive value with path: ${path.slice(1).join('.')}`)
    }

    return get(next, path.slice(1))
}
```